pdl/PDL/map_fluxon_flow.pdl

```diff
@@ -1,88 +1,197 @@
 
-=head2 map_fluxon_flow
+=head1 AUTHOR
 
-=for ref
+Gilly <gilly@swri.org>
+Chris Lowder
 
-Given a world and list of fluxons, generates a mapping of the solar wind flow along these fluxon structures.
+=head1 LICENSE
+
+This library is free software; you can redistribute it and/or modify it under the same terms as Perl itself.
 
 =cut
 
+=head2 map_fluxon_flow_parallel_master
 
+=over
 
-sub map_fluxon_flow {
-    my $before = clock_gettime();
+=item B<Description>
 
-    my $fn = shift;
-    my $fluxons = shift;
-    my @fluxons = @{$fluxons};
+Given a world and list of fluxons, this function generates a mapping of the solar wind flow along these fluxon structures. It uses parallel processing to speed up the calculations.
 
+=item B<Parameters>
 
+=over
 
+=item C<$file_name>: The name of the output file where the results will be saved.
 
-# Generate some blank storage arrays
-my @fps = ();   # Fluxon array position
-my @phb = ();   # Begining coordinates
-my @thb = ();
-my @phe = ();   # End coordinates
-my @the = ();
-my @vrb = ();   # Beginning and end wind velocity
-my @vre = ();
-my @frb = ();   # Beginning and end expansion factors
-my @fre = ();
-my @fre2 = ();
+=item C<$fluxons>: A reference to an array containing the fluxons to be mapped.
 
-# Loop through open fluxons and generate wind profiles
-for my $fid(0..scalar(@fluxons)-1){
+=item C<$max_processes>: (Optional) The maximum number of parallel processes to use. Defaults to the value of C<$concurrency>.
 
+=back
 
-    # if ($fid>0){
-    # print "\r", '  Calculating fluxon:', $fid, '/', scalar(@fluxons);
-    # }
-    my $me = $fluxons[$fid];
+=item B<Returns>
 
-    # Check for open fieldlines
-    my $st_open = ($me->{fc_start}->{label}==-1);
-    my $en_open = ($me->{fc_end}->{label}==-2);
-    if ($me->{plasmoid} || ($st_open + $en_open != 1)){
+The function writes the results to disk and returns nothing.
 
-        next; 
-    }
+=back
 
-    # Find the transonic wind solution
-    (my $farr, my $fr, my $bth, my $bph) = gen_fluxon_tflow($me);
-
-    # Append any needed values and move along home
-    # Note to distinguish between fluxons ends...
-    push(@fps, $fid);
-    push(@phb, squeeze($bph(0,0)));
-    push(@thb, squeeze($bth(0,0)));
-    push(@phe, squeeze($bph(0,1)));
-    push(@the, squeeze($bth(0,1)));
-    push(@vrb, $farr(1,0));
-    push(@vre, $farr(1,-1));
-    push(@frb, $fr(1,1));
-    push(@fre, $fr(-2,1));
-    push(@fre2, $fr(-1,1));
+=cut
 
-    my $after   = clock_gettime();
-    my $elapsed = $after - $before;
-    my $round_elapsed = rint($elapsed*10) / 10;
-    print "\r", '  Calculated fluxons:', $fid, '/', scalar(@fluxons), ", ", $round_elapsed, "(s) elapsed";
-    
+sub map_fluxon_flow {
+    my $file_name = shift;
+    my $fluxons = shift;
+    my @fluxons = @{$fluxons};
+    my $max_processes = shift || 6;    # Set the number of parallel processes
```

Comment:
Aha, I replaced a bit of code before the function declaration that was hardcoding it to 11 threads. I moved it inside and set a lower default value if not otherwise specified. For most base ARM-based macs these days the number of performance cores is 6, so I thought that'd be alright.